services/api/src/resources/task/resolvers.ts

```diff
@@ -125,7 +126,30 @@ export const getTasksByEnvironmentId: ResolverFn = async (
     queryBuilder = queryBuilder.limit(limit);
   }
 
-  return query(sqlClientPool, queryBuilder.toString());
+  const rows = await query(sqlClientPool, queryBuilder.toString())
+
+  let adminTasks = false
+  // do a check of all returned tasks for any that are admin tasks
+  for (const row of rows) {
+    if (row.adminOnlyView == true) {
+      adminTasks = true
```

Comment:
Although this is in memory, so it'll be nice and fast, you could "break" here, since the first instance of an admin task tells us that there are admin tasks.